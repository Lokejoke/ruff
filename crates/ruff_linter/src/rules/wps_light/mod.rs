//! Rules from [wps-light](https://pypi.org/project/wps-light/).
pub(crate) mod rules;

#[cfg(test)]
mod tests {
    use std::convert::AsRef;
    use std::path::Path;

    use anyhow::Result;
    use test_case::test_case;

    use crate::registry::Rule;
    use crate::test::test_path;
    use crate::{assert_messages, settings};

    #[test_case(Rule::ListMultiplication, Path::new("WPS435.py"))]
    #[test_case(Rule::ReservedArgumentAsVariable, Path::new("WPS117.py"))]
    #[test_case(Rule::UnderscoresInNumber, Path::new("WPS303.py"))]
    #[test_case(Rule::AssignmentToSubscriptSlice, Path::new("WPS362.py"))]
    fn rules(rule_code: Rule, path: &Path) -> Result<()> {
        let snapshot = format!("{}_{}", rule_code.as_ref(), path.to_string_lossy());
        let diagnostics = test_path(
            Path::new("wps_light").join(path).as_path(),
            &settings::LinterSettings::for_rule(rule_code),
        )?;
        assert_messages!(snapshot, diagnostics);
        Ok(())
    }
}
