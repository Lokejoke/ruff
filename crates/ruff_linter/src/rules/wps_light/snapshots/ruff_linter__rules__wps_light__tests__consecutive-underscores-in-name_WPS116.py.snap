---
source: crates/ruff_linter/src/rules/wps_light/mod.rs
snapshot_kind: text
---
WPS116.py:27:1: WPS116 Variable name _private__variable contains consecutive underscors inside.
   |
25 | # incorect
26 | 
27 | _private__variable = 1
   | ^^^^^^^^^^^^^^^^^^ WPS116
28 | 
29 | CONSTANT__VARIABLE = 1
   |
   = help: Rename variable to _private_variable

WPS116.py:29:1: WPS116 Variable name CONSTANT__VARIABLE contains consecutive underscors inside.
   |
27 | _private__variable = 1
28 | 
29 | CONSTANT__VARIABLE = 1
   | ^^^^^^^^^^^^^^^^^^ WPS116
30 | 
31 | def function__name():
   |
   = help: Rename variable to CONSTANT_VARIABLE

WPS116.py:31:5: WPS116 Function name function__name contains consecutive underscors inside.
   |
29 | CONSTANT__VARIABLE = 1
30 | 
31 | def function__name():
   |     ^^^^^^^^^^^^^^ WPS116
32 |     in_function__variable = 1
   |
   = help: Rename variable to function_name

WPS116.py:32:5: WPS116 Variable name in_function__variable contains consecutive underscors inside.
   |
31 | def function__name():
32 |     in_function__variable = 1
   |     ^^^^^^^^^^^^^^^^^^^^^ WPS116
33 | 
34 | long__variable___name = 1
   |
   = help: Rename variable to in_function_variable

WPS116.py:34:1: WPS116 Variable name long__variable___name contains consecutive underscors inside.
   |
32 |     in_function__variable = 1
33 | 
34 | long__variable___name = 1
   | ^^^^^^^^^^^^^^^^^^^^^ WPS116
35 | 
36 | class ClassName:
   |
   = help: Rename variable to long_variable_name

WPS116.py:37:9: WPS116 Function name __dunder___method__ contains consecutive underscors inside.
   |
36 | class ClassName:
37 |     def __dunder___method__(self, parameter_x):
   |         ^^^^^^^^^^^^^^^^^^^ WPS116
38 |         pass
   |
   = help: Rename variable to __dunder_method__
